### ISSUE TITLE
🐛 `useImportType` false positive when identifier shares a name with a TypeScript type

### ISSUE THREAD

**Issue by ahrjarrett** [biomejs/biome#4640]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.11.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.7.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

This one's easy to reproduce.

With `useImportType`* enabled, biome treats certain term-level identifiers (such as `object`) as types, even if the identifier is used as a value in that file.

Since the `unsafe` flag is opt-in, this has the potential to cause a runtime exception unless it's caught during review / by our CI/CD pipeline.

Config:

I believe `useImportType` is turned on by default, so any configuration that doesn't override it is enough to reproduce.

Repro:

```typescript
/// ex-01.ts
export namespace object {
  export const keys = globalThis.Object.keys
}
```

This is what I wrote:

```typescript
/// ex-02.ts
import { object } from "./ex_01.js"

console.log(object.keys(globalThis))
```

This is what biome changes it to:

```typescript
/// ex-02.ts

//       vvvv
import { type object } from "./ex_01.js"

//            vv this now throws
console.log(object.keys(globalThis))
```

### Expected result

It shouldn't add inline type keywords if the identifier is used as a term in that same file.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I can't reproduce it in the playground, this means it's been fixed. Can you try to latest nightly to confirm?

**Comment by ivan-kleshnin** [comment]
I'm experimenting with the latest version (1.9.4)

```ts
import React, {useRef, useState} from "react"
```

is suggested to be replaced with

```ts
import type React from "react"
import {useRef, useState} from "react"
```

by `lint/style/useImportType` rule. But **react** is gonna be imported into this module anyway (not to mention it's a framework that gonna be bundled anyway).

**Comment by ematipico** [comment]
Going to close the issue since OP hasn't replied.

@ivan-kleshnin please check the documentation of the rule, it provides suggestions for React: https://biomejs.dev/linter/rules/use-import-type/#options
